flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/ExprCodeGenerator.scala

```diff
@@ -476,6 +485,11 @@ class ExprCodeGenerator(ctx: CodeGeneratorContext, nullableInput: Boolean)
             call.getOperator.getReturnTypeInference == ReturnTypes.ARG0 =>
         generateNullLiteral(resultType)
 
+      // We only support JSON function operands within JSON_OBJECT
+      case (operand: RexNode, _)
```

Comment:
```suggestion
      case (operand: RexNode, i)
```
also check that the index is even. otherwise JSON() function could be used for the key?